wadas/ui/configure_tunnel.py

```diff
@@ -0,0 +1,142 @@
+        self.ui.lineEdit_tunnel_name.textChanged.connect(self.validate)
+        self.ui.comboBox_camera_1.currentIndexChanged.connect(self.on_camera1_combobox_idx_changed)
+        self.ui.comboBox_camera_1.currentIndexChanged.connect(self.on_camera1_combobox_idx_changed)
+        self.ui.pushButton_direction_camera_1.clicked.connect(self.set_camera1_direction)
+        self.ui.pushButton_direction_camera_2.clicked.connect(self.set_camera2_direction)
+
+        if self.tunnel:
+            self.ui.lineEdit_tunnel_name.setText(tunnel.id)
+            self.ui.comboBox_camera_1.setCurrentText(tunnel.camera_entrance_1)
+            self.ui.label_direction_camera_1.setText(tunnel.entrance_1_direction.value)
+            self.ui.comboBox_camera_2.setCurrentText(tunnel.camera_entrance_2)
+            self.ui.label_direction_camera_2.setText(tunnel.entrance_2_direction.value)
+
+    def initialize_camera1_combobox(self):
+        """Method to initialize camera list dropdown with available cameras."""
+
+        self.ui.comboBox_camera_1.clear()
+        for camera in self.cameras_not_in_tunnels:
+            if self.ui.comboBox_camera_2.currentText() != camera:
+                self.ui.comboBox_camera_1.addItem(camera)
+
+    def initialize_camera2_combobox(self):
+        """Method to initialize camera list dropdown with available cameras."""
+
+        self.ui.comboBox_camera_2.clear()
+        for camera in self.cameras_not_in_tunnels:
+                if self.ui.comboBox_camera_1.currentText() != camera:
+                    self.ui.comboBox_camera_2.addItem(camera)
+
+    def on_camera1_combobox_idx_changed(self):
+        """Update camera 2 options when camera 1 changes."""
+        self.initialize_camera2_combobox()
+        self.validate()
+
+    def on_camera2_combobox_idx_changed(self):
+        """Update camera 1 options when camera 2 changes."""
+        self.initialize_camera1_combobox()
+        self.validate()
+
+    def set_camera1_direction(self):
+        """Method to set camera 1 direction"""
+
+        if (dlg := DialogConfigureCameraForTunnelMode(self.ui.label_direction_camera_1.text())).exec():
+            self.ui.label_direction_camera_1.setText(dlg.direction.value)
+        self.validate()
+
+    def set_camera2_direction(self):
+        """Method to set camera 2 direction"""
+
+        if (dlg := DialogConfigureCameraForTunnelMode(self.ui.label_direction_camera_2.text())).exec():
+            self.ui.label_direction_camera_2.setText(dlg.direction.value)
+        self.validate()
+
+    def validate(self):
+        """Method to make sure all input are provided before clicking OK button"""
+
+        valid = (bool(self.ui.lineEdit_tunnel_name.text()) and
+                 bool(self.ui.comboBox_camera_1.currentText()) and
+                 bool(self.ui.comboBox_camera_2.currentText()) and
+                 bool(self.ui.label_direction_camera_1.text()) and
+                 bool(self.ui.label_direction_camera_2.text()))
```

Comment:
valid = bool(self.ui.lineEdit_tunnel_name.text() and
                 self.ui.comboBox_camera_1.currentText() and
                 self.ui.comboBox_camera_2.currentText() and
                 self.ui.label_direction_camera_1.text() and
                 self.ui.label_direction_camera_2.text())